sigma/filters.py

```diff
@@ -0,0 +1,191 @@
+import random
+import re
+import string
+from dataclasses import dataclass, field
+from typing import List, Optional, Union
+from uuid import UUID
+
+import yaml
+
+from sigma import exceptions as sigma_exceptions
+from sigma.correlations import SigmaCorrelationRule
+from sigma.exceptions import SigmaRuleLocation
+from sigma.rule import (
+    SigmaYAMLLoader,
+    SigmaLogSource,
+    SigmaDetections,
+    SigmaDetection,
+    SigmaRule,
+    SigmaRuleBase,
+)
+
+
+class SigmaFilterLocation(sigma_exceptions.SigmaRuleLocation):
+    """Location of Sigma filter in source file."""
+
+    pass
+
+
+@dataclass
+class SigmaGlobalFilter(SigmaDetections):
+    rules: List[UUID] = field(default_factory=list)
+
+    @classmethod
+    def from_dict(
+        cls, detections: dict, source: Optional[SigmaRuleLocation] = None
+    ) -> "SigmaGlobalFilter":
+        try:
+            if isinstance(detections["condition"], list):
+                condition = detections["condition"]
+            else:
+                condition = [detections["condition"]]
+        except KeyError:
+            raise sigma_exceptions.SigmaConditionError(
+                "Sigma rule must contain at least one condition", source=source
+            )
+
+        try:
+            if isinstance(detections["rules"], list):
+                rules = detections["rules"]
+            else:
+                rules = [detections["rules"]]
+        except KeyError:
+            raise sigma_exceptions.SigmaConditionError(
+                "Sigma rule must contain at least one condition", source=source
```

Comment:
```suggestion
                "Sigma rule must contain at least a rules section", source=source
```
Why not add a generic `SigmaGlobalFilterError` Class